apps/console/src/features/applications/components/settings/attribute-management/attribute-selection-oidc.tsx

```diff
@@ -0,0 +1,943 @@
+
+    /**
+     * Handle change event of the search input.
+     *
+     * @param event - Change event.
+     */
+    const handleChange = (event) => {
+        const changeValue = event.target.value;
+
+        setSearchValue(changeValue);
+        if (changeValue.length > 0) {
+            searchFilter(changeValue);
+        } else {
+            setExpandedScopes([]);
+            setExternalClaimsGroupedByScopes(unfilteredExternalClaimsGroupedByScopes);
+        }
+    };
+
+    /**
+     * Set initial values for set local dialect values.
+     */
+    const setInitialValues = () => {
+        setUserAttributesLoading(true);
+
+        const initialRequest = getInitiallySelectedClaimsURI();
+        const initialSelectedClaims: ExtendedExternalClaimInterface[] = [];
+        const initialAvailableClaims: ExtendedExternalClaimInterface[] = [];
+
+        applicationConfig.attributeSettings.attributeSelection.getExternalClaims(externalClaims)
+            .map((claim: ExternalClaim) => {
+                if (initialRequest.includes(claim.mappedLocalClaimURI)) {
+                    const newClaim: ExtendedExternalClaimInterface = {
+                        ...claim,
+                        mandatory: checkInitialRequestMandatory(claim.mappedLocalClaimURI),
+                        requested: true
+                    };
+
+                    initialSelectedClaims.push(newClaim);
+
+                } else {
+                    initialAvailableClaims.push(claim);
+                }
+            });
+
+        const tempFilterSelectedExternalClaims = [ ...filterSelectedExternalClaims ];
+
+        claimConfigurations?.claimMappings?.map((claimMapping: ClaimMappingInterface) => {
+            claims?.map((claim) => {
+                if (claimMapping.localClaim.uri === claim.claimURI){
+                    const option: ExtendedExternalClaimInterface = {
+                        claimDialectURI: claim.dialectURI,
+                        claimURI: claim.claimURI,
+                        id: claim.id,
+                        localClaimDisplayName: claim.displayName,
+                        mandatory: claim.mandatory,
+                        mappedLocalClaimURI: claim.claimURI,
+                        requested: claim.requested
+                    };
+
+                    if (!initialSelectedClaims.find(
+                        (selectedExternalClaim) => selectedExternalClaim?.mappedLocalClaimURI
+                        === claimMapping.localClaim.uri)){
+                        initialSelectedClaims.push(option);
+                    }
+                }
+            });
+        });
+
+        setFilterSelectedExternalClaims(tempFilterSelectedExternalClaims);
+        setSelectedExternalClaims(initialSelectedClaims);
+        setAvailableExternalClaims(initialAvailableClaims);
+        setExternalClaims(initialAvailableClaims.concat(initialSelectedClaims));
+        setInitializationFinished(true);
+    };
+
+    useEffect(() => {
```

Comment:
Use effect should occur before functions.

Reply:
resolved with https://github.com/wso2/identity-apps/pull/3566/commits/66c501e3bafdfeba484d242847e688d8d0ee2611